libs/base/ui/structure/heading/src/typography.controller.ts

```diff
@@ -37,28 +37,52 @@ export class TypographyController implements ReactiveController {
   }
 
   protected setStyle(tag?: HeadingTag | HeadingVisibility, size?: Size): void {
+    this.setFont(tag, size);
+    this.setDisplay(tag, size);
+    this.setTransform(tag, size);
+  }
+
+  /**
+   * Sets the font properties for the heading, based on the provided tag and size.
+   */
+  protected setFont(tag?: HeadingTag | HeadingVisibility, size?: Size): void {
     if (!tag) return;
 
     this.setStyleProperty('--_s', size, `var(--oryx-typography-${tag}-size)`);
     this.setStyleProperty('--_w', size, `var(--oryx-typography-${tag}-weight)`);
     this.setStyleProperty('--_l', size, `var(--oryx-typography-${tag}-line)`);
+  }
 
-    this.setStyleProperty(
-      '--_t',
-      size,
-      tag === HeadingTag.Subtitle ? `uppercase` : undefined
-    );
-
-    if (size) {
+  /**
+   * Sets the display property for the heading, based on the provided tag and size.
+   * When no tag is provided, the default display is `inline`.
+   * When no size is provided, the default display is `inline-block`.
+   *
+   */
+  protected setDisplay(
+    tag?: HeadingTag | HeadingVisibility,
+    size?: Size
+  ): void {
+    if (!tag || (!size && this.isInlineStyle(tag))) {
```

Comment:
Is the description for this method correct? Description says when no tag -> display: 'inline', but according the implementation - it is `inline-block`